Bug 1973636 - Add isNewUntil Search Definition to Rust Engine Selector.

mandysGit opened the pull request:
### Pull Request checklist ###
<!-- Before submitting the PR, please address each item -->
- **Breaking changes**:  This PR follows our [breaking change policy](https://github.com/mozilla/application-services/blob/main/docs/howtos/breaking-changes.md)
  - [ ] This PR follows the breaking change policy:
     - This PR has no breaking API changes, or
     - There are corresponding PRs for our consumer applications that resolve the breaking changes and have been approved
- [ ] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/releases.md) after merging.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGELOG.md](../CHANGELOG.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due diligence applied in selecting them.

Standard8 reviewed (changes requested):
Thank you, this is looking good, but there's a couple of extra bits to add:

* Please add an entry to `CHANGELOG.md`. Now that we are being used by multiple apps, we should be adding entries there. 
* Please can we extend `test_from_configuration_details_merges_variants` and `test_from_configuration_details_merges_sub_variants` to test that `is_new_until` gets properly merged.

mandysGit commented:
Addressed Review comments. 

After this lands, when may I land my other patch: https://phabricator.services.mozilla.com/D253104 ? 
Would I need to wait until this code gets vendored into Mozilla Central first?

Standard8 reviewed (approved):
Thank you, this looks good. For landing, it'll need to be vendored across. 

I'll check to see if anyone is planning to do a vendor soon.
